ansible_base/lib/abstract_models/common.py

```diff
@@ -161,8 +161,11 @@ def from_db(self, db, field_names, values):
 
         for field in self.encrypted_fields:
             field_value = getattr(instance, field, None)
-            if field_value:
-                setattr(instance, field, ansible_encryption.decrypt_string(field_value))
+            if field_value and type(field_value) is str and field_value.startswith("$encrypted$"):
+                decrypted_field_value = ansible_encryption.decrypt_string(field_value)
+                if isinstance(self._meta.get_field(field), models.JSONField):
+                    decrypted_field_value = json.loads(decrypted_field_value)
+                setattr(instance, field, decrypted_field_value)
```

Comment:
Same thing with json.loads here. Also note, that decrypt_string will not auto check the value and return it directly so I don't think we need any changes in here anymore.